Modules/Devices/src/Devices.Application/Devices/Commands/DeleteDevice/Handler.cs

```diff
@@ -23,18 +21,19 @@ public Handler(IIdentitiesRepository identitiesRepository, IUserContext userCont
 
     public async Task Handle(DeleteDeviceCommand request, CancellationToken cancellationToken)
     {
-        var device = await _identitiesRepository.GetDeviceById(request.DeviceId, cancellationToken, track: true);
+        var deviceThatIsDeletingId = _userContext.GetDeviceId();
+        var deviceThatIsDeleting = await _identitiesRepository.GetDeviceById(deviceThatIsDeletingId, cancellationToken, track: true);
 
-        if (device.Identity.Address != _userContext.GetAddress())
-        {
-            throw new NotFoundException(nameof(device));
-        }
+        if (deviceThatIsDeleting.Identity.Address != _userContext.GetAddress())
+            throw new NotFoundException(nameof(deviceThatIsDeleting));
 
-        _logger.LogTrace("Challenge successfully validated.");
+        var deviceThatIsBeingDeleted = await _identitiesRepository.GetDeviceById(request.DeviceId, cancellationToken, track: true);
 
-        device.MarkAsDeleted(request.DeviceId);
+        if (deviceThatIsBeingDeleted.Identity.Address != _userContext.GetAddress())
+            throw new NotFoundException(nameof(deviceThatIsBeingDeleted));
```

Comment:
As this check is moved to the `Device` class this is obsolete. In case of the check failing the `DomainError` is returned there, as per the method signature:
```csharp
    public DomainError? CanBeDeleted(DeviceId deletedByDevice, IdentityAddress identityAddress)
    {
        if (IsOnboarded)
            return new DomainError("error.platform.validation.device.deviceCannotBeDeleted", "The device cannot be deleted because it is already onboarded.");

        if (IdentityAddress != identityAddress)
            return new DomainError("error.platform.validation.device.deviceCannotBeDeleted", "The device cannot be deleted because it does not belong to the current user.");

        return null;
    }
```